enclave-interface/README.md

```diff
@@ -0,0 +1,369 @@
+# - Feature Name: enclave-interface
+- Start Date: 2020-02-11
+- RFC PR: (leave this empty)
+- Ursa Issue: (leave this empty)
+- Version: 0.1
+- Contributors: Mike Lodder, Jon Geater, Mick Bowman, Carsten Stöcker, Hart Montgomery, Brent Zundel
+
+# Summary
+[summary]: #summary
+Hardware security modules(HSM)/Trusted Execution Environments(TEE)/Secure enclaves are becoming more common for specialized cryptography key management.
+This RFC describes a common API that can be used for interacting with enclaves in Ursa. This API allows for
```

Comment:
Is this API intended to be usable outside of Rust?

If so, may want to separate the abstract API from a concrete implementation in Rust... also might be easier to align with other things if that were the case.

Reply:
Thanks for reviewing this. I agree that the API should be abstract. That’s why I’ve designed it this way. I do say while the concrete example here is in Rust, the API could easily be adapted to any other language. Rust also compiles to WASM and C. I fully intend for this to be as widely adoptable and portable as possible.